src/test/java/org/bonitasoft/connectors/scripting/ShellConnectorTest.java

```diff
@@ -11,182 +11,192 @@
 
-import org.bonitasoft.connectors.scripting.ShellConnector;
 import org.bonitasoft.engine.connector.ConnectorException;
 import org.bonitasoft.engine.connector.ConnectorValidationException;
-import org.junit.Ignore;
 import org.junit.Test;
 
 /**
  * @author Hongwen Zang
- *         -Dbonita.home=${workspace_loc}/bos-continuous-integration-tests/bos-local-continuous-integration-tests/target/home -DsmtpServerAddress=192.168.1.211
- *         -DsmtpServerPort=555
  */
 public class ShellConnectorTest extends ConnectorTest {
 
-    private String getInterpreter() {
-        return isUnix() || isMac() ? "/bin/sh" : "cmd.exe";
+    private static final Logger LOGGER = Logger.getLogger(ShellConnectorTest.class.getName());
+
+    @Test
+    public void should_run_multi_line_script() throws Exception {
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("interpreter", defaultOsInterpreter());
+        parameters.put("parameter", defaultOsParameter());
+        parameters.put("script", multiLineShellScript());
+
+        ShellConnector shell = validateAndExecute(parameters);
+        assertThat((String) shell.getResult()).isNotEmpty();
     }
 
-    private String getParameter() {
-        return isUnix() || isMac() ? "-c" : "/c";
+    @Test
+    public void should_list_content_of_current_directory() throws Exception {
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("interpreter", defaultOsInterpreter());
+        parameters.put("parameter", defaultOsParameter());
+        parameters.put("script", listContentOfCurrentDirectoryShellScript());
+
+        ShellConnector shell = validateAndExecute(parameters);
+        assertThat((String) shell.getResult()).isNotEmpty();
     }
 
-    private String getShellScript() {
-        return isUnix() || isMac() ? "ls -lia" : "dir";
+    @Test
+    public void should_exit_with_non_zero_status() throws Exception {
+        final Map<String, Object> parameters = new HashMap<>();
+        parameters.put("interpreter", defaultOsInterpreter());
+        parameters.put("parameter", defaultOsParameter());
+        parameters.put("script", "exit 3");
+
+        ShellConnector shell = validateAndExecute(parameters);
+        assertThat(shell.getExitStatus()).as("exit status").isEqualTo(3);
     }
 
-    private String getMultiLineShellScript() {
-        if (isUnix() || isMac()) {
-            return "ls \n"
-                    + "var=2"
-                    + "echo $var \n"
-                    + "var=$(($var+1)) \n"
-                    + "echo var"
-                    + "sleep 2";
-        }
-        else {
-            return "dir \n"
-                    + "set var=2 \n"
-                    + "echo %var% \n"
-                    + "set /a var=%var%+1 \n"
-                    + "echo %var% \n"
-                    + "ping 123.45.67.89 -n 1 -w 2500 > nul \n";
+    @Test
+    public void should_run_sleep_script() throws Exception {
```

Comment:
I am not sure what this method really tests?

Reply:
The test previously existed. I have just renamed it